pkg/util/kubeversion/kubeversion_test.go

```diff
@@ -0,0 +1,80 @@
+package kubeversion
+
+import (
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	versionutil "k8s.io/apimachinery/pkg/util/version"
+	"k8s.io/apimachinery/pkg/version"
+
+	fakediscovery "k8s.io/client-go/discovery/fake"
+	fakeclientset "k8s.io/client-go/kubernetes/fake"
+	kubetesting "k8s.io/client-go/testing"
+)
+
+// FetchServerVersion gets API server version
+func TestFetchServerVersion(t *testing.T) {
+	cases := map[string]struct {
+		ticker    *time.Ticker
+		version   string
+		wantEqual *versionutil.Version
+	}{
+		"server version equals expected version": {
+			version:   "v1.0.0",
+			wantEqual: versionutil.MustParseGeneric("v1.0.0"),
+		},
+	}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			fakeClient := fakeclientset.NewSimpleClientset()
+			fakeDiscovery, ok := fakeClient.Discovery().(*fakediscovery.FakeDiscovery)
+			if !ok {
+				t.Fatalf("couldn't convert Discovery() to *FakeDiscovery")
+			}
+			fakeDiscovery.FakedServerVersion = &version.Info{
+				GitVersion: tc.version,
+			}
+			_ = NewServerVersionFetcher(fakeDiscovery, tc.ticker).FetchServerVersion()
+			if serverVersion.String() != tc.wantEqual.String() {
+				t.Errorf("Unexpected result, want %v", tc.wantEqual)
+			}
+		})
+	}
+}
+
+func TestFetchServerVersionWithError(t *testing.T) {
```

Comment:
This test looks like an overkill. I don't expect this to fail.

Reply:
I think the test is added as I pointed out in the initial version, there would be nil version set, and Kueue would crash on panic checking `AtLeast` on nil. While unlikely, the version check may happen in case of transient connectivity issues. Also, the API server under heavy load may respond with 429 (too many requests). The test execution time is negligible, so I think it makes sense to keep it, but no strong opinion.

Reply:
I agree, the test was added for that purpose, but anyway if it seems overkill I can delete it